### PR TITLE
[#13]: grn

### PR DESCRIPTION
**Pull request by goodylove**
Trello: https://trello.com/c/fLekjH5n/13-grn

### CODE REVIEW

File: cubeseed_login/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "next": "13.3.0",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "react-responsive-modal": "^6.4.2",
```
Comment (abu-hasib) [inline cubeseed_login/package.json:17]: Please consult with the tech stack here https://docs.google.com/document/d/1otSVVp7G5-K1eNmDUV5-jHWlj8JHG9KH3xgwZGicbBg/edit before adding a new library.

Comment (abu-hasib) [reply cubeseed_login/package.json:17]: Did you make changes to all 80 files?

File: cubeseed_login/src/pages/_app.tsx [hunk 2]
```diff
@@ -1,6 +1,11 @@
-import type { AppProps } from 'next/app'
-import "@/styles/globals.scss"
+import type { AppProps } from "next/app";
+import "@/styles/globals.scss";
+import { FarmContexProvider } from "@/context/context";
 
 export default function App({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  return (
+    <FarmContexProvider>
```
Comment (abu-hasib) [inline cubeseed_login/src/pages/_app.tsx:7]: Why did you wrap the root component in this?

File: cubeseed_login/src/pages/login_page/login-page.tsx [hunk 3]
```diff
@@ -11,46 +11,44 @@
...
   const [password, setPassword] = useState("");
   const [rememberMe, setRememberMe] = useState(false);
-  const [emailError, setEmailError] = useState('');
-  const [passwordError, setPasswordError] = useState('');
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
   const router = useRouter();
 
   const handleLogin = () => {
-    if (email === '') {
```
Comment (abu-hasib) [inline cubeseed_login/src/pages/login_page/login-page.tsx:44]: i suggest using a lib for forms, https://react-hook-form.com/ is a good one.

File: cubeseed_login/src/pages/login_page/login-page.tsx [hunk 4]
```diff
@@ -80,30 +77,30 @@ const LoginPage: React.FC<LoginFormProps> = ({
     login(email, password);
   };
 
-  const login = async(username: string, password: string) => {
+  const login = async (username: string, password: string) => {
     try {
       // api url
-      const response = await fetch('http://localhost:8000/api/auth/token/', {
```
Comment (abu-hasib) [inline cubeseed_login/src/pages/login_page/login-page.tsx:86]: did you work on this page

File: cubeseed_login/package.json (hunk 1 above)
Comment (goodylove) [reply cubeseed_login/package.json:17]: no ..the last person did not commit there work thats why

File: cubeseed_login/src/pages/login_page/login-page.tsx (hunk 3 above)
Comment (goodylove) [reply cubeseed_login/src/pages/login_page/login-page.tsx:44]: i did not work on this

File: cubeseed_login/src/pages/login_page/login-page.tsx (hunk 4 above)
Comment (goodylove) [reply cubeseed_login/src/pages/login_page/login-page.tsx:86]: no

File: cubeseed_login/src/pages/_app.tsx (hunk 2 above)
Comment (goodylove) [reply cubeseed_login/src/pages/_app.tsx:7]: i needed  to use a state in a different component ...so lifting it up was the only option for now .

File: cubeseed_login/package.json (hunk 1 above)
Comment (goodylove) [reply cubeseed_login/package.json:17]: i have check the tech  stack ...cant i use any other library?

**Review by MimaoQiuQiu — APPROVED** [review]
Thanks for your work

**Pull request merged**